vpc/service/branch_enis.go

```diff
@@ -1124,11 +1136,40 @@ func (vpcService *vpcService) createBranchENI(ctx context.Context, tx *sql.Tx, s
 	ctx, span := trace.StartSpan(ctx, "createBranchENI")
 	defer span.End()
 
+	// TODO: Use idempotency token
+	// TODO: Get rid of ENI Backfill
+	row := tx.QueryRowContext(ctx, `
+SELECT subnet_usable_prefix.prefix
+FROM subnet_usable_prefix
+JOIN subnets ON subnet_usable_prefix.subnet_id = subnets.id
+JOIN subnet_cidr_reservations_v6 ON subnet_usable_prefix.subnet_id = subnet_cidr_reservations_v6.subnet_id
+AND subnet_usable_prefix.prefix <<= subnet_cidr_reservations_v6.prefix
+WHERE subnets.subnet_id = $1
+  AND subnet_usable_prefix.branch_eni_id IS NULL
+  AND subnet_cidr_reservations_v6.description = $2
+  AND subnet_cidr_reservations_v6.type = 'explicit'
+  FOR
+  NO KEY UPDATE OF subnet_usable_prefix SKIP LOCKED
+  LIMIT 1
+`, subnetID, vpcService.subnetCIDRReservationDescription)
+	var prefix string
+	err := row.Scan(&prefix)
+	if err != nil {
+		err = fmt.Errorf("Not able to find usable prefix in subnet to assign to interface: %w", err)
+		tracehelpers.SetStatus(err, span)
+		return nil, err
+	}
+
 	createNetworkInterfaceInput := ec2.CreateNetworkInterfaceInput{
-		Ipv6AddressCount: aws.Int64(0),
-		SubnetId:         aws.String(subnetID),
-		Description:      aws.String(vpcService.branchNetworkInterfaceDescription),
-		Groups:           aws.StringSlice(securityGroups),
+		// Ipv6AddressCount: aws.Int64(0),
```

Comment:
Hmm this is weird, what is this number "supposed" to be when prefixes are involved?